ansible/playbooks/roles/upgrade/tasks/kubernetes.yml

```diff
@@ -1,25 +1,67 @@
 ---
+# During HA control plane upgrade server address in kubeconfig is switched to local for
+#   * compatibility between client and server versions
+#   * identifying correct server version
+
+- name: k8s/master | Switch apiserver address to local
+  include_tasks: kubernetes/utils/set-local-apiserver.yml # sets kubectl_context_cluster
+  when:
+    - groups.kubernetes_master | length > 1
+    - inventory_hostname in groups.kubernetes_master
+
 - name: Wait for kube-apiserver and get cluster version
-  delegate_to: "{{ groups.kubernetes_master[0] }}"
+  delegate_to: >-
+    {{ inventory_hostname if inventory_hostname in groups.kubernetes_master else
+      groups.kubernetes_master[0] }}
   block:
     - name: k8s | Include wait-for-kube-apiserver.yml
       import_tasks: kubernetes/utils/wait-for-kube-apiserver.yml
 
     - name: k8s | Include get-cluster-version.yml
-      import_tasks: kubernetes/get-cluster-version.yml # sets cluster_version
+      import_tasks: kubernetes/get-cluster-version.yml
+
+    - name: Set initial_cluster_version fact # sets initial_cluster_version
+      set_fact:
+        initial_cluster_version: >-
+          {{ (kubectl_cluster_version.stdout | from_yaml).serverVersion.gitVersion }}
+
+    - name: k8s/kubelet-cm | Include set-cluster-version.yml
```

Comment:
Not sure if we need this task here. If `cluster_version` is used somewhere before `patch-kubelet-cm.yml`, then I would just add the fact to the previous task.

Reply:
We are using all of the facts (`version`, `major_version` and `minor_version`) that we are setting in set-cluster-version.yml -> Therefore it would be a code duplication if we will set it here manually in previous task.